src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clone = { ...state };
+  const change = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(clone, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete clone[key];
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const key in clone) {
+        delete clone[key];
+      }
+    }
+    change.push({ ...clone });
```

Comment:
Ensure that the spread operator `{ ...clone }` is necessary here. It creates a shallow copy of the `clone` object, which is important if the state objects need to be independent of each other in the `change` array.